src/widgets/NavigationBar.js

```diff
@@ -8,38 +8,78 @@ import PropTypes from 'prop-types';
 import { Platform, TouchableOpacity, View, StyleSheet } from 'react-native';
 import Icon from 'react-native-vector-icons/FontAwesome';
 import { textStyles } from '../globalStyles';
+import { BadgeSet } from './BadgeSet';
 
-export function NavigationBar({ onPressBack, LeftComponent, CentreComponent, RightComponent }) {
-  return (
-    <View style={localStyles.container} >
-      <View style={localStyles.leftSection}>
-        <TouchableOpacity onPress={onPressBack} style={localStyles.backButton}>
-          {onPressBack && <Icon name={'chevron-left'} style={localStyles.backIcon} />}
-        </TouchableOpacity>
-        {LeftComponent && <LeftComponent />}
-      </View>
-      <View style={localStyles.centreSection} >
-        {CentreComponent && <CentreComponent />}
-      </View>
-      <View style={localStyles.rightSection}>
-        {RightComponent && <RightComponent />}
+export class NavigationBar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
Could be 1 line.

Reply:
Is it actually necessary to do this at all? Should be able to just write the whole constructor off if I understand react correctly. That said, I'd understand the value of stating what state you expect to be populated in this component:
```javascript
export class NavigationBar extends React.Component {

state = {
  unfinalisedCount: null,
}
// ...
```